sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/sampleoverview/SampleOverviewView.groovy

```diff
@@ -54,9 +67,12 @@ class SampleOverviewView extends VerticalLayout implements HasHotbar, HasTitle,
         ListDataProvider<Sample> dataProvider = ListDataProvider.ofCollection(samples)
         samplesGrid.addColumn(Sample::getCode).setCaption("Sample Code").setId("SampleCode")
         samplesGrid.addColumn(Sample::getName).setCaption("Sample Name").setId("SampleName")
-        samplesGrid.addColumn(Sample::getStatus).setCaption("Sample Status").setId("SampleStatus").setStyleGenerator({Sample sample -> determineColor(sample.status)})
+        samplesGrid.addColumn(Sample::getStatus).setCaption("Sample Status").setId("SampleStatus")
+                .setStyleGenerator({Sample sample -> determineColor(sample.status)})
         samplesGrid.setSelectionMode(Grid.SelectionMode.NONE)
         samplesGrid.setDataProvider(dataProvider)
+        dataProvider.addFilter({ sampleFilter.asPredicate().test(it) })
+        dataProvider.refreshAll()
```

Comment:
maybe it would make sense to add the filter along with the combobox? Or we create a method to add filters that gets the dataprovider as parameter. I think we have multiple filters and with this, we could sort the code a little.

Reply:
My idea was that there is only one filter that is configured in different places 😄